### DISCUSSION TITLE
Changes in InvenTree require manual refresh for implementation

### DISCUSSION THREAD

**Discussion by palagap** [inventree/InvenTree#7491] [Q&A]
Hi there,

When making changes in Inventree, they are not always apparent until a manual refresh of the webpage. This can cause issues such as allocating more than the required number of parts to a build order.  Without a manual refresh of the page, it is not possible to be sure that the part was allocated correctly.  An example of this issue using the Inventree Demo (0.16.0 dev) is shown below:

1. Build Order as viewed prior to expanding the Build Output and allocating parts:

![Inventree Refresh 1](https://github.com/inventree/InvenTree/assets/40808444/39e5c984-143d-46aa-9b05-6848873c4406)

2. Build Order after expanding the Build Output:

![Inventree Refresh 2](https://github.com/inventree/InvenTree/assets/40808444/a642344e-56d5-4291-8a30-6675a4c840b5)

3. Clicking the “Allocate” button:

![Inventree Refresh 3](https://github.com/inventree/InvenTree/assets/40808444/1574a258-c004-4eb1-a388-86e5718c001d)

4. Selecting a part and quantity to allocate:

![Inventree Refresh 4](https://github.com/inventree/InvenTree/assets/40808444/32e7a9a6-4e18-4514-87fc-772adcb91089)

5. After allocating a part and clicking “Submit”, I am brought back to this page.  For the part, K-testcomponent-3 | K test component, the “Allocated” column should indicate that 1 of 3 parts has been allocated.  Instead, it appears as though no parts have been allocated.  To fix this, the page has to be refreshed:

![Inventree Refresh 5](https://github.com/inventree/InvenTree/assets/40808444/0f4de824-be0e-4ac8-ba4e-a2032fb6a948)

6. After the page is refreshed, the Build Output is minimized again:

![Inventree Refresh 6](https://github.com/inventree/InvenTree/assets/40808444/d50df907-f8ad-432e-939c-ad9735f66f4d)

7. Finally, after expanding the Build Output, the parts appear allocated properly:

![Inventree Refresh 7](https://github.com/inventree/InvenTree/assets/40808444/d6e3cbb8-8d08-478d-9e29-44a2cd142c42)

I would appreciate any insight on how to resolve this issue. 

Thanks.

**Comment by SchrodingersGat** [comment]
There are a number of issues with the "responsiveness" of the existing UI. We are currently engaged in a [major overhaul of the user interface](https://github.com/inventree/InvenTree/issues/5212) to address this, and many other issues / limitations.

Once the UI is migrated to react, then auto-update of page data will be significantly better. You can view the progress at https://demo.inventree.org/platform/

Note that we are a limited developer team and any assistance or sponsorship of our efforts to modernize the UI are greatly appreciated, and will help us get it released sooner.

**Reply by palagap** [reply]
Sounds good, thanks for the update.
